### PR TITLE
TF-11504 add description to projects

### PR DESCRIPTION
**Pull request by netramali**
## Description

Description attribute is added to projects in atlas. 

_Remember to:_

- [x] _Update the [Change Log](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/changelog-process.md)_
- [x] _Update the [Documentation](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/changelog-process.md#updating-the-documentation)_

## Testing plan

Integration tests

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [Related PR](https://github.com/hashicorp/atlas/pull/18568)
- [Related PR](https://github.com/hashicorp/atlas/pull/18494)
- [go-tfe]( https://github.com/hashicorp/go-tfe/pull/861)
- [Jira](https://hashicorp.atlassian.net/browse/TF-11504)

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See [testing.md](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/testing.md) to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
=== RUN   TestAccTFEProject_basic
--- PASS: TestAccTFEProject_basic (6.15s)
PASS

=== RUN   TestAccTFEProject_update
--- PASS: TestAccTFEProject_update (6.52s)
PASS

datasource:
=== RUN   TestAccTFEProjectDataSource_basic
--- PASS: TestAccTFEProjectDataSource_basic (8.07s)
PASS
```
Screenshot 2024-03-01 at 9 55 15 AM

### CODE REVIEW

**Review by Maed223 — DISMISSED** [review]
Looks good overall 👍, just the CHANGELOG that needs adjustment

File: internal/provider/data_source_project.go [hunk 1]
```diff
@@ -28,6 +28,11 @@ func dataSourceTFEProject() *schema.Resource {
 				Required: true,
 			},
 
+			"description": {
```
Comment (laurenolivia) [inline internal/provider/data_source_project.go:31]: @netramali if you're adding an attribute to a data source we should note this in the changelog - see [example](https://github.com/hashicorp/terraform-provider-tfe/blob/main/CHANGELOG.md#v0500)

Comment (netramali) [reply internal/provider/data_source_project.go:31]: I got a suggestion to remove it. I have added it back again.

**Review by hs26gill — APPROVED** [review]
LGTM!

**Pull request merged**